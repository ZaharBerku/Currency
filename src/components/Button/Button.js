import React from "react";
import PropTypes from "prop-types";

const Button = (props) => {
    const {text, handelClick, additionalClassNames, children} = props;
    return(
        <button onClick={handelClick} className={additionalClassNames}>{text || children}</button>
    );
};

Button.defaultProps = {
    handelClick:()=>{},
    additionalClassNames:"",
    children:null,
    text:null
};

Button.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]),
    text: PropTypes.string,
    handelClick: PropTypes.func,
    additionalClassNames: PropTypes.string,
};


export { Button };