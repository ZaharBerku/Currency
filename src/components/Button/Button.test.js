/* eslint-disable no-undef */
import React from "react";
import { Button } from "./Button";
import { render } from "@testing-library/react";

const handelClick = jest.fn();

const Component = (props) => (
  <Button additionalClassNames="active" {...props}>
    Click me
  </Button>
);

describe("Rendor button", () => {
    test("should render button", () => {
      const { asFragment } = render(<Component/>);
      expect(asFragment()).toMatchSnapshot();
    });
});

describe("Button click", () => {
    test("should click on button work", () => {
      const { getByTestId } = render(<Component handelClick={handelClick}/>);
      const button = getByTestId("button");
      button.click();
      expect(handelClick).toBeCalled();
    });
});