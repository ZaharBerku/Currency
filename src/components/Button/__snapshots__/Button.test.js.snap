// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendor button should render button 1`] = `
<DocumentFragment>
  <button
    class="active"
    data-testid="button"
  >
    Click me
  </button>
</DocumentFragment>
`;
